strawberryfields/engine.py

```diff
@@ -356,6 +356,8 @@ def run(self, program, *, shots=1, compile_options={}, modes=None, state_options
             Result: results of the computation
         """
 
+        # session or feed_dict are needed by TF backend during simulation if program contains measurements
+        kwargs.update(state_options)
```

Comment:
Note: I needed to add this to ensure correct functionality of TF backend. It's not enough to pass these during the call to `state`